### ISSUE TITLE
TensorQTL result from "ALL" celltype only analysis and the 8 tissue analysis are different.

### ISSUE THREAD

**Issue by hsun3163** [cumc/xqtl-pipeline#271]
In the analysis with "ALL" cell type, one of the nominal results of TensorQTL is as followed:
```
hs3163@node55:/mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/Association_scan_arch_2/ALL/TensorQTL$ cat ALL.log2cpm.bed.chr1.norminal.cis_long_table.txt | grep "19190" | grep "HES4"

phenotype_id    variant_id      tss_distance    af      ma_samples      ma_count        pval    beta    se      alt     ref     pos     chrom
HES4    chr1:19190_GC_G -980982 0.047927458     37      37      0.7408239165888748      0.030455409     0.092000164     G       GC      19190   chr1
```

However, the same gene-snp pair produced a different nominal result
```
HES4    chr1:19190_GC_G -980982 0.047927458     37      37      0.8287406868015537      -0.020334253    0.09392069      G       GC      19190   chr1
```

**Comment by hsun3163** [comment]
I believed the difference is caused by the difference in factor, as in the new analysis, the factor, generated by PEER with residY is 
```
```

**Comment by hsun3163** [comment]
The aforementioned difference is in turns caused by the differences in PCA,
```
hs3163@node55:/mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis$ zcat /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/output/data_preprocessing/ALL/covariates/ALL.covariate.pca.gz | tail -1
PC13    -2.1920833776991914e-4  0.06185181061249991     -0.06179998598727744    0.08151105197510006     0.0021634715750909754   -0.09188383421131945    -0.0682454910220896     0.03931432350466356     0.07019941765270449     -0.024027209535655156   0.003983776235706505  0.0028495970775288663    0.012710543509204761    -0.053799344796856656   0.0703537545951703      0.006569329847921626    0.022265507767507058    0.054613896977250366    0.12928116261799866     -0.0329255850879568     -0.039219232923058844   -0.05717834229959189    0.054679164315119      0.05728341004495634     0.020062179793052987    0.02828861290506565     0.0678387665925067      0.044445610765492444    0.035687818549703855    0.01982588251518323     -0.004772737232683357   0.07759989306467995     -0.08937102491223607    -0.04654685090651912   0.06674096746134271     -2.371601788787803e-4   -0.005787836684672586   0.08498520139464832     0.02297476836020163     -0.02881445970570314    0.002080148964591458    -0.037685112120323556   0.03911781530582127     0.0042815354649434045   0.01083905739030702   0.010739293519154697     -0.03588078387761555    0.02104900255309978     -0.07520581623431837    0.04893904144341179     -0.08997613604440648    -0.05281301102320535    -0.023909467894603176   0.03821940530782634     5.048155943630582e-4    0.007798697964116597    0.08011473992777911    9.67183110761495e-5     0.02452071890990509     0.012120857118774149    0.1252516062573643      0.06035000302886074     -0.10559613509756687    0.005177163306015929    -0.1356162203107816     0.010373647423746022    0.07141424171760284     0.04236795581366652    -0.08870082857785845    -0.1459157814603937     -0.028149951585852862   -0.03715646969660829    0.017969132597950785    0.0029473803383842233   -0.029932056581683002   -0.06287123623235018    -0.016348193248723414   -0.014476536759805954   0.022922529143451352  0.022766743222250693     0.1064130487761984      -0.005761083100838012   0.0025655047093322367   0.03222771657379196     0.015073721462849276    0.020811504935705778    -0.008053549258923319   -0.054564223771195075   0.09939047332998512     -0.023448642965005505   -0.01310525334675958   -0.015870523077406624   -0.023662218836968552   -0.09508251523634806    0.007460065964430063    -0.012112930881899303   0.008407168844536472    0.058031301888765506    0.06358101265809739     -0.1231709881989314     -0.05652589640674737    -0.07606237313343764   -0.09433591255672863    0.04617153462121992     0.0020498981306796608   1.13208899774762e-4     0.026989219208850508    0.01525038780311027     -0.09694056994313699    -0.11193261942513838    -0.011821086458415649   0.02834917797412823     -0.05114443231493094  0.017446854855235446     -0.05577994986659821    -0.002558037345421795   0.09012823799595998     -0.03342819795752653    -0.02508974928360538    -0.029399302420433326   -0.04068122405108389    -0.04116081647704578    0.06305642925685372     -0.034205654010426814   0.0512375691040727     0.06786895854209282     0.0536907418404607      0.04404991219563322     -0.0161320018410668     0.02566924538843323     0.04118292947968881     6.351113188613951e-4    -0.023635884572846155   0.0036426745896466816   0.015385446129733039    -0.017079930631046562  0.03855392315222051     0.01860481773359427     -0.09472954244201706    -0.012268834644067163   -0.05275663004670923    -0.05564603212929333    0.027899254831063088    -0.018318775155872796   0.04405317675434774     -0.00653080172233011    -0.056908023203018666 -0.05044684877951665     -0.019612978369935398   -0.02642237861585442    0.03885173482355365     -0.08523399049100464    0.034663518255832235    0.06379419494303548     0.08452530298089021     0.054446399394240605    -0.0138097425591511     0.029629959899396725    -0.0354440552160201    -0.05297155591565021    -0.0689279072890397     0.11376769217532777     0.0059922554939928424   -0.07654933761559601    0.1334157958216009      0.08097880797903437     -0.009963869144968126   -0.035487110922076424   2.096556264762088e-5    -0.04366425441743307   -0.07286444541689147    0.057844311020157364    -0.056155203844491      -0.06253736340790181    -0.08570209786610622    0.04214299708751736     0.08752501787665018     0.10642791073176347     -0.05220440489045933    0.07175843762992402     -0.01608257101761936  0.048443943975514124     -0.026663848007975773   0.03906281754609583     -0.014249538828830993   -0.034886476172151126   0.11932175358515085     -0.019448939587748716   -0.08674283866698047    -0.03462492956636089    -0.07345812481382993    0.02775250958197465     -3.056655465775439e-4  0.0017474223133887316   0.02766780259014556     -0.06051943210300513    0.005008578867936592    -5.831794321652316e-4   3.027823373819714e-4    0.004028067743975916    -0.058389813604835555   0.04163407094079434     -0.005854501651651757   0.0927516592441538     0.050748980467611525    -0.00426944430413701    -0.07140642550457806    -0.041016493044267605   0.043954416871836634    0.016254932996544805    -0.018122747463412196   0.05092966898787975     -0.00271873085753663    0.01290447787736405     -0.011566090429940017 -0.021437432436483533    -0.059241544458911825   -0.002920793370937275   0.016503546683568262    -0.051693662966086344   0.03958490740309814     -0.009108737656053096   -0.06523774125270254    0.0027465810869181755   0.04607412841762837     0.046419630885865926    0.0063082716835939695  -0.009395695008024913   0.04102175237232623     0.04544034799815306     0.0035106686116329854   0.08226426397672212     0.05363790765386664     0.007506183277499738    -0.021192188128174812   0.049622239964633504    -0.01833006474191556    -0.0676274640785661    -0.02915794523659486    0.0607199541427645      -0.035754558203219565   -0.023079740969679262   -0.08765765241365163    0.020039807360377183    -0.023422839398523396   0.0318308012078612      0.057523529595888225    0.043977133796727366    -0.018436630250461473 -0.010031260121971053    0.05869797827623316     0.025290979572057958    0.13475659455020395     0.050658606126537095    -0.010736070406552192   -0.03743648529042087    0.07379857173664368     -0.04196912318580159    -0.02796472965825709    -0.056462793801410206   -0.020787728819055613  -0.034684232185074895   0.006631169927920318    -0.06697647476410122    0.0395385632586065      -0.00621423260441838    8.672186913320528e-5    0.02368435361313145     -0.060651982854497064   0.028783404557472934    -0.004229364445313246   0.017434969173270757   -0.03792123601282895    -0.008033371693791072   0.008480446107028942    0.013851955898196823    -0.09051169892784777    0.0694366915386837      0.07035892445346213     -0.07327897236608746    0.024158969940573757    -0.04254178586147925    -0.007468693121946308 0.02691023453039953      0.010067251184976447    -0.042070976341939556   0.05397367451941898     -0.010540669009193735   0.08011356667310052     -0.02866001976251259    0.03992854653388408     -0.014359804075292712   -0.038514704483826864   0.08978293356383175     -0.004860661927863924  0.10437248780317279     -0.08210517249073604    0.026310677948447724    -0.04962410053518263    -0.027359085101537332   -9.23297006014285e-4    -0.06105820003772649    -0.018753337536406185   -0.04040846832277845    -0.03383796076731815    0.05596091365136595    0.06248985364750023     0.09458867590381691     0.026829277177654404    0.039062849282417295    0.06508719884115638     0.0820985791297892      -0.04862125028983215    -0.0633210702603132     0.05043343762881341     -0.06160450755304435    -0.04264353532028017  0.06826566660460812      -0.08623673949016575    0.015180894314481273    -0.012341175618606053   -0.007643930736203653   -0.06107860396903766    0.04253640527606636     -0.02425588515463259    0.002901221734916139    -0.04331083100106938    0.0998598588177963      -0.015019492027986841  -0.05310064659250895    -0.067581949648072      -0.0014255931912970357  0.0863863788014661      -0.14332476756624757    0.017900313551821763    0.06368802708621202     -0.03185869937508206    0.01816873637031748     0.010982246410182884    0.02837436357779781    0.0645021008292171      0.026213086270775617    0.05359048235363853     0.08584366815031923     0.03452450110751499     0.05779736611007198     0.06712357972515785     -0.09554324158705117    -0.02365744068240169    -0.018049799183102448   -0.027623322134685182 -0.03660944468611735     -0.009900056376642595   -0.0816451026617381     -0.05967032348990772    -0.06961933136439148    -0.023161518450426716   0.04108917841815657     -0.017618402774684527   0.07173127748285359     -0.05712359610794931    -0.012934243786498843   0.005137494097978055   0.03836493396822379     -0.044579589095301575   -0.11393161606387268    0.04177484201492178     -0.07137292628283204    -0.029001983274139956   -0.054848402368633566   -0.0024772745615648135  -0.025025361073550627   0.10429953328727642     -0.05028144567760915   -0.023991468015122055   0.021177297512180237    0.04308857119752351     -0.01898225873007233    -0.0257797211608735     0.0265269172626284      0.009958318304441242    0.061517594068374225    0.018844927271967984    0.07248337698427873     0.027290448899578493  -0.06699367546719845     0.02595043657733336     -0.06653831643772581    0.039806570309730296    -0.0051360191639238375  0.01492069713687997     0.076875925399961       0.0053849455425770245   -0.037126779272976765   0.01800982260371215     -0.0017425843401047193  -0.01710048527914623   0.09332444152819966     -0.08607064868448461    0.01755985234092635     -0.02117440330814451    -0.08715628878608205    -0.0765343067956247     -0.11504175856586546    0.02907949170830555     -0.1073439985715357     0.07126153040072178     0.012783988366445941   -0.03258596178935651    0.03203788377588585     0.03142048283480275     0.017544555655340095    -0.005792152059287534   -0.02744456442325256    -0.014070639705670834
hs3163@node55:/mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis$ zcat /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/ata_preprocessing_arch/ALL/covariates/ALL.covariate.pca.gz | tail -1
gzip: /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/ata_preprocessing_arch/ALL/covariates/ALL.covariate.pca.gz: No such file or directory
hs3163@node55:/mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis$ zcat /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/data_preprocessing_arch/A | tail -1
ALL/ Ast/
hs3163@node55:/mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis$ zcat /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/data_preprocessing_arch/A | tail -1
ALL/ Ast/
hs3163@node55:/mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis$ zcat /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/data_preprocessing_arch/ALL/covariates/ALL. | tail -1
ALL.covariate.pca.BiCV.cov.gz           ALL.covariate.pca.cov.PEER.cov.gz       ALL.log2cpm.bed.fake.vcf.gz             ALL.log2cpm.resid.PEER.cov.gz
ALL.covariate.pca.PEER.cov.gz           ALL.covariate.pca.cov.gz                ALL.log2cpm.resid.PEER.cov.PEER.cov.gz
hs3163@node55:/mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis$ zcat /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/data_preprocessing_arch/ALL/covariates/ALL.covariate.pca.cov.gz | tail -1
PC13    0.006876095448411422    0.0333537806393171      0.04941953738730537     -0.016036982153298823   -0.08224306022744368    -0.040807865484672065   -0.02481004019190205    -0.015476581527739956   0.017460172184669723    -0.02498541766516367    -0.018957682537467546 0.07535044072186535      -0.00443033436883562    -0.024441824988557546   0.04664417139946503     -0.09712157818170661    -0.07631066734093532    0.05158010645198147     0.04386471458587423     0.04166944568226979     -0.02935881625247036    -0.01694673438520922    -0.05687953914693016   -8.859562059953987e-4   -0.04216075802029023    0.004409689367086875    0.046854427134005734    -0.003242949244366134   0.005567554176081906    0.044452109354656794    0.09219995187915281     0.006501665084407706    -0.10435155620913324    -0.06893933483532966   -0.1053071296610665     0.029660875507717406    -0.04060177924345829    -0.054776078024879125   0.019035006499255866    0.02203215512843258     0.10664041578028662     -0.09306045236903772    0.0384786285317504      -0.05906594404502413    -0.011550426867614143 0.0997752572221123       0.053551045455235015    -0.005437222001117659   0.014153664347102168    -0.023834445912907783   0.021127397831016144    0.05575062654758172     -0.020718727216344223   -0.08864048160301724    -0.02004066109811205    0.08089556191627609     0.005884021949283206   -0.031901475208293555   -0.09492034591801336    0.0845584504323265      -0.06760415248544813    -0.029378967103418935   0.019910651077897246    -0.015818426947299246   -0.017072702779768238   -0.006734379654508568   -0.011209216359166262   -0.09562888248820026   0.08867170288214453     -0.045130851174271146   0.012546325186717628    -0.07516790059665115    -0.04351634281664174    0.019250012441738303    -0.06398127068337517    0.0794728952984035      0.08037698896342393     -0.006460221020113661   -0.05728280106606293  0.11239619135434607      -0.010305204673391782   -0.10209420314250649    -0.012221678164945977   -0.0017417870746121535  0.03449512829247896     -0.12446737737952221    0.024331153986941047    0.07166375568947748     0.01603527744047716     -0.08066576309972699    -0.016800733204258057  -0.06639034577775808    0.048855511469247256    -0.005534267746811807   0.003871246811865066    0.01960246611499271     -0.01785142708327759    0.012111618032747266    -0.049980431600067404   0.07432904644520451     0.005695000871904004    -0.04177915098586113   -0.03575183668451113    -0.13581098634984642    0.02234535494383354     -0.05323319517051423    -0.05929672772215481    0.04249557206736403     -0.0015643489511602128  -0.003913406974070977   -0.03620257778790558    0.03693116302278537     0.006561967461465693  -0.10129838717250672     -0.00418611042453345    -0.018037670256308355   0.005876349472676844    -0.0069582464775420445  -0.0039354726130499266  -0.01862403403160427    -0.05775790851520233    0.09040256090985756     -0.014337773013392953   0.07892612183313902     -0.030288369230929937  -0.06795822935351012    -0.0077217355124800215  0.008684715926771846    0.03504109369919855     -0.016468490112011035   -0.006921987101517854   -1.8082789494333997e-4  -0.08553074994775538    0.08436072126389209     0.01569723077874748     -0.023421750255025874  0.1146292067923218      -0.03339664327675631    0.06464617199426793     0.06854421639981938     0.10633689815706725     0.017470078529302664    -0.0756456834097288     -0.0037127013389480117  -0.020979264300004297   0.017834272485335805    -0.06043100872551908  0.039329778524525155     0.038176377799471334    0.02216619022287088     -2.3896652162785461e-4  0.05701693412637939     0.017483989338325498    -0.04758801575777629    -0.03170735067070451    0.03275104124650552     0.022402797502023526    -0.05081888426953422    0.006036423704934573   0.0018357534508947147   0.07196555991851096     -0.0046775075160445925  -0.06538038284310145    0.026778009725504103    0.004585876345506788    -0.010573787527756957   0.09006916579858233     -0.0011651112942368323  0.03344034466886108     0.011663699693196647   0.03146487341672898     -0.015767909563297124   0.029622442400664223    0.024538986489442707    -0.07318989748413221    0.09018841707761707     0.09447801164779177     0.02711148658990552     0.1658752090187987      0.002483893028785427    0.05152276009020335   0.005028742737121955     0.05347316030201176     0.06632777354664875     -0.02834701895082771    0.015820846143167266    -0.04176418384228515    -0.1000638632805503     -0.09099981013653745    0.14175486552996114     -0.03401913230800413    -0.08456237204356973    0.040271096739041654   -0.06385641577305123    -0.06037250661459844    -0.02034187446764552    -0.03991683140312391    0.04099602168655031     0.02293085797288223     -0.0360438948981804     -0.015359751021080788   -0.055855953547123395   0.04235664612660776     -0.059924536970039     -5.183169411454996e-4   0.06132010774509924     -0.0053033471511930615  -0.08694582149430385    -0.04278602752055128    -0.01581585458877153    0.014449790534294895    0.07405000292213604     0.016350823971515258    -0.023132675894668765   0.0571074402058472    0.11769356559477928      0.007215405565969564    -0.06722078663083579    0.04081132770783686     -0.04153335810224343    -0.06247567949330376    -0.046636674277358334   0.08150353574567473     0.07181947207320047     0.006667525360149326    0.08342569898973021     -0.04351659118713017   0.004382622719620672    -0.007616257399475017   -0.07364136687126198    0.020735620492762695    -0.04709125068629372    0.009903552854734557    0.027124816867069128    -0.05330231766737899    0.05798930236889663     0.03342092220150468     0.03351616587839946    -0.018282684611211176   -0.033044514522712246   -0.06204849092829135    0.10831184389328526     -0.02288426978075411    0.018470013236207714    -0.039328429634621305   0.010560798358928215    0.01921040983168873     0.02822477650210558     -0.015291178234198663 0.013359756762377136     -0.007758844871734702   0.056143978927372766    -0.0020482358826894266  0.021420446064079175    0.050754859033592635    0.01607746308624654     0.009473825531388102    -0.004983715672022635   0.07106856072166405     -0.023378150878784802   -0.05643102701722679   0.04297756218537614     0.12592585220143127     -0.09972489086830101    -0.040745590100898184   -0.05767233372976221    0.042508077073330805    0.048075722366256686    -0.04255401137486426    0.05027908283252804     -0.01814249519120582    0.007677889478797509   0.0063699120585613426   0.004571078321731147    0.00494874627943943     0.017037754417251       0.03762007635495771     -0.04782054776018467    0.007705208018590836    0.03197510460762896     0.00726530713309873     -0.127977209858454      0.019465694385844563  -0.07183603589728124     -0.018443352214567405   -0.0030091735417508716  0.0573545946576136      0.06031212151992136     -0.027829691937002386   0.07163355897794012     0.05053171137055577     0.048180608605115624    -0.022643552147143434   0.05799609903129835     -0.059763376804602295  0.09457616490025829     0.06435180923484886     -0.01263162673735126    -0.1292452671587238     -0.018774157801628527   0.022038016133842767    0.031344428271839544    0.01082537219426514     0.00552791587645241     0.011576296985498918    -0.07778630894198185   0.02218738984358293     0.01826073279006882     0.02835617931245671     -0.08054883950311424    0.01880978277964652     -0.02301358778082564    -0.062381885441800636   -0.0053214008854988335  -0.05461058991762183    -0.03284865741269614    0.002886321405156788  0.06726645964520667      0.020070753186183794    -0.0020807085695993743  0.01835259428184773     -0.002389973251028334   -0.014662521308216277   -0.025930462017751895   -0.052463465516134004   0.03476109329615781     -0.020114258287500193   -0.1421593196171796     0.050410608345980344   -0.021592841636690827   -0.03536316167147307    0.03600448289624392     0.13677204967797932     0.06145139830389908     -0.005236892823890664   0.08248149188934085     0.04835968783277058     -0.01960151013177487    0.014591143408841588    -0.05971064715164602   -0.06506192827038225    -0.006881450659375415   -0.035543204649922173   -0.04791740255539785    -0.005992800736936452   0.03803696622351991     -0.08593303063814944    -0.0064042530504284665  0.024898352081910494    -0.04743150252671794    -0.07971750017014173  -0.0122374538222214      -0.003608326175357289   -0.04980612708390826    -0.02412861717185786    -0.040498225695963334   -0.06199862027497434    0.013813822721326391    0.08871152681493627     -0.01518780114256422    0.05011746713635204     -0.010723981027755637   -0.03568426986958431   -0.0015060575618907118  0.09977529912114619     0.058555065150232455    0.020442010238816873    -0.0494958710188558     -0.014496536255106532   -0.09154850249894607    0.033685842953921984    -0.048996669137279784   0.07143674810153566     0.028333352655085183   -0.026229391768438158   0.06986537606521309     -0.022426145735827192   -0.07817688191326841    0.08483241207493754     -0.026766249667126867   -0.05859705384675344    0.012473131840461456    -0.052049418160376816   0.08087329503868698     0.058425231528902825  -0.049994345532059344    -0.02954496490884573    -0.04242490418109253    0.04796587461542579     -0.09055792844952569    -0.0028666845491823684  -0.02603566407285442    0.052491352009192666    0.09568235744853093     0.008875658439666837    -0.0019719286764167425  -0.041533512115169244  0.016718698254720585    0.03666759338372656     0.039561520118663096    0.03821011516669744     0.05701782138381416     0.03427894563753651     0.028221166210266435    0.0058568759490823475   -0.06442701770191823    0.08674670049147412     -0.04166382257029995   -0.016232966267910855   0.02066773243851799     0.020062946587783743    0.03752921561298849     -0.011619373270312755   -0.09770097519636747    0.004927354602085639
```


which is then caused by the difference of related ID determined by king:
```
hs3163@node55:/mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis$ cat /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/data_preprocessing_arch/ALL/genotype_data/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.related_id
SM-CJK5A SM-CJK5A
hs3163@node55:/mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis$ cat /mnt/mfs/statgen/snuc_pseudo_bulk/eight_tissue_analysis/output/data_preprocessing/ALL/genotype_data/GRCh38_liftedover_sorted_all.add_chr.leftnorm.filtered.renamed.filtered.renamed.filtered.ALL.related_id
SM-CJIZF SM-CJIZF
SM-CJK5A SM-CJK5A
```

Which was implemented according to #264 

The impact of this difference on the number of significant genes .etc is unclear yet.
